src/dsql/parse.y

```diff
@@ -4737,14 +4737,16 @@ array_range
 %type <legacyField> simple_type
 simple_type
 	: non_charset_simple_type
-	| character_type charset_clause
+	| character_type charset_clause collate_clause
 		{
```

Comment:
I'm not sure whether it works in our codebase, but accordingly to the SQL specification `collate_clause` must also be supported for text blobs.

Reply:
Yes, but Firebird has no standard character blobs (CLOB or CHARACTER LARGE OBJECT), only binary blobs so this part is not for this PR.